WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListViewModel.kt

```diff
@@ -91,6 +98,35 @@ class NotificationsListViewModel @Inject constructor(
         }
     }
 
+    fun openNote(
+        noteId: String?,
+        openInTheReader: (siteId: Long, postId: Long, commentId: Long) -> Unit,
+        openDetailView: () -> Unit
+    ) {
+        val note = noteId?.let { notificationsUtilsWrapper.getNoteById(noteId) }
+        if (note != null && note.isCommentType && !note.canModerate()) {
+            val readerPost = readerPostTableWrapper.getBlogPost(note.siteId.toLong(), note.postId.toLong(), false)
+            if (readerPost != null) {
+                openInTheReader(note.siteId.toLong(), note.postId.toLong(), note.commentId)
+            } else {
+                readerPostActionsWrapper.requestBlogPost(
+                    note.siteId.toLong(),
+                    note.postId.toLong(),
+                    object : ReaderActions.OnRequestListener<String> {
+                        override fun onSuccess(result: String?) {
+                            openInTheReader(note.siteId.toLong(), note.postId.toLong(), note.commentId)
+                        }
+
+                        override fun onFailure(statusCode: Int) {
+                            openDetailView()
```

Comment:
Is failure expected in some cases? Wondering whether we should add some kind of log here. 🤔

Reply:
This will happen if the reader post is not present locally and the call to fetch it fails. I don't expect this to happen that often and the fallback is defaulting to the current behaviour of loading the detail view (which I guess might fail too with an error). I'll add some extra logging here though there is logging on the [api call failure](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java#L293).

Reply:
Added with b2d02dd9b18174e0f756f56865455611cda9ebe2

Reply:
Thanks for explaining!